MdePkg/Include/Library/HobLib.h

```diff
@@ -204,6 +204,32 @@ BuildResourceDescriptorWithOwnerHob (
   IN EFI_GUID                     *OwnerGUID
   );
 
+/**
+  Builds a HOB that describes a chunk of system memory with memory attributes.
+
+  This function builds a HOB that describes a chunk of system memory.
+  It can only be invoked during PEI phase;
+  for DXE phase, it will ASSERT() since PEI HOB is read-only for DXE phase.
+
+  If there is no additional space for HOB creation, then ASSERT().
+
+  @param  ResourceType        The type of resource described by this HOB.
+  @param  ResourceAttribute   The resource attributes of the memory described by this HOB.
+  @param  PhysicalStart       The 64 bit physical address of memory described by this HOB.
+  @param  NumberOfBytes       The length of the memory described by this HOB in bytes.
+  @param  Attributes          The memory attribute for the memory described by this HOB.
+
+**/
+VOID
+EFIAPI
+BuildResourceDescriptorWithCacheHob (

```

Comment:
```suggestion
BuildResourceDescriptorWithEfiMemoryAttributesHob (

```

Maybe something besides this, but trying to be a bit clearer on what this is creating, WithCache both is confusing (are you caching the hob, etc.) and maybe too restrictive, it is cache attributes for now, could be extended in the future.

Reply:
Is the WIthCacheHob attempting to say that only coachability attributes are being reported, or that the hob is in cache?

Does this also encompass the bits covered by EFI_MEMORY_ACCESS_MASK and EFI_MEMORY_ATTRIBUTE_MASK?

Reply:
Updated the name to be v2. For the current implementation, C core will not act on these hobs different than the v1 hobs.